File: Projetos/ACBrLib/Demos/C#/Shared/ACBrLib.Core/Config/ACBrLibConfigBase.cs
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace ACBrLib.Core.Config
{
    public abstract class ACBrLibConfigBase<TLib> where TLib : ACBrLibHandle
    {
        #region Fields

        protected readonly TLib Parent;
        protected readonly ACBrSessao SessaoConfig;
        protected string SubName;

        #endregion Fields

        #region Constructors

        protected ACBrLibConfigBase(TLib acbrlib, ACBrSessao sessao)
        {
            Parent = acbrlib;
            SessaoConfig = sessao;
        }

        #endregion Constructors

        #region Methods

        protected TProp GetProperty<TProp>([CallerMemberName] string propertyName = null)
        {
            return Parent.ConfigLerValor<TProp>(SessaoConfig, string.IsNullOrEmpty(SubName) ? propertyName : $"{SubName}.{propertyName}");
        }

        protected void SetProperty<TProp>(TProp newValue, [CallerMemberName] string propertyName = null)
        {
            Parent.ConfigGravarValor(SessaoConfig, string.IsNullOrEmpty(SubName) ? propertyName : $"{SubName}.{propertyName}", newValue);
        }

        #endregion Methods
    }
}

File: Projetos/ACBrLib/Demos/Java/ACBr Common/src/com/acbr/ACBrLibConfigBase.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.acbr;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 *
 * @author rften
 * @param <TLib>
 */
public abstract class ACBrLibConfigBase<TLib extends ACBrLibBase> {
    protected TLib parent;
    protected ACBrSessao sessaoConfig;
    protected String subName;
    
    protected ACBrLibConfigBase(TLib acbrlib, ACBrSessao sessao){
        parent = acbrlib;
        sessaoConfig = sessao;
    }
       
    private String getPropertyName(String propertyName){
        if ("".equals(subName))
            return propertyName;
        else
            return subName + "." + propertyName;
    }
    
    protected String getProperty(String propertyName) throws Exception{
        return parent.configLerValor(sessaoConfig,  getPropertyName(propertyName));         
    }
    
    protected void setProperty(String propertyName, String value) throws Exception{
        parent.configGravarValor(sessaoConfig,  getPropertyName(propertyName), value);        
    }
    
    protected int getIntProperty(String propertyName) throws Exception{
        String value = getProperty(propertyName);
        return Integer.parseInt(value);
    }
    
    protected void setIntProperty(String propertyName, int value) throws Exception{
        String svalue = String.valueOf(value);
        setProperty(propertyName, svalue);
    }
    
    protected boolean getBoolProperty(String propertyName) throws Exception{
        String value = getProperty(propertyName);
        return Boolean.parseBoolean(value);
    }
    
    protected void setBoolProperty(String propertyName, boolean value) throws Exception{
        String svalue = value ? "1" : "0";
        setProperty(propertyName, svalue);
    }
    
    protected Date getDateProperty(String propertyName) throws Exception{
        String value = getProperty(propertyName);
        SimpleDateFormat formato = new SimpleDateFormat("dd/MM/yyyy"); 
        return formato.parse(value);
    }
    
    protected void setDateProperty(String propertyName, Date value) throws Exception{
        SimpleDateFormat formato = new SimpleDateFormat("dd/MM/yyyy"); 
        String svalue = formato.format(value);
        setProperty(propertyName, svalue);
    }
}
